### PR TITLE
Feature/Login

### PR DESCRIPTION
**Pull request by Prakort**
image

### CODE REVIEW

File: app/javascript/components/Login/validation.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const LoginValidation = (form) => {
+  let error = {};
+
+  const { email, password} = form;
```
Comment (amanpatel123) [inline app/javascript/components/Login/validation.js]:
```suggestion
  const { email, password } = form;
```

File: app/javascript/components/Login/validation.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  if(!email) error.email = "Invalid Email!";
+  if(!password) error.password = "Invalid password!";
+
+  return {
+    valid: Object.keys(error).length === 0,
```
Comment (amanpatel123) [inline app/javascript/components/Login/validation.js:10]: What does this object refer to?

File: app/javascript/components/Login/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    confirmPassword: ''
+  })
+
+  const handleSubmit = (event) => {
+    // prevent page from refreshing    
+    event.preventDefault();
+    setLoading(true);
+    // validate the form
+    setErrors({});
+    const error = LoginValidation(form);
```
Comment (amanpatel123) [inline app/javascript/components/Login/Login.jsx:23]: I like how you handled validation in another file

File: app/javascript/components/Login/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    if(!error.valid){
+      setErrors(error);
+      setLoading(false);
+    } else {
+
+      //TODO: api call with backend to sign up the user
```
Comment (amanpatel123) [inline app/javascript/components/Login/Login.jsx:30]: Make sense!!

**Comment by amanpatel123** [comment]
@Prakort I think we have a decision to make here. Why do we have Drivers login here? Why will a driver use a web dashboard for?

File: app/javascript/components/Login/validation.js (hunk 2 above)
Comment (Prakort) [reply app/javascript/components/Login/validation.js:10]: This object refers to the errors of each field and we could store the customed message. `valid` for checking if there is no error in any fields then it should be `true` such that we don't need to check again.

**Comment by Prakort** [comment]
> @Prakort I think we have a decision to make here. Why do we have Drivers login here? Why will a driver use a web dashboard for?

Let's just build the web for the driver for now. Mobile app for the driver requires more meetings.

**Comment by amanpatel123** [comment]
> Let's just build the web for the driver for now. Mobile app for the driver requires more meetings.
@Prakort  I think let's concentrate on what businesses will see for now. Making a web for drivers does not really make sense since the phone is handy and can be used while delivering.

**Comment by Prakort** [comment]
> > Let's just build the web for the driver for now. Mobile app for the driver requires more meetings.
> > @Prakort  I think let's concentrate on what businesses will see for now. Making a web for drivers does not really make sense since the phone is handy and can be used while delivering.

okay, we need to have a meeting and discuss this. If we have the sign-up for driver, we should have a platform for the driver to login. Maybe a progressive web app?

**Comment by amanpatel123** [comment]
@Prakort Lets build the Business view first and not worry about the Drivers view

**Review by amanpatel123 — APPROVED** [review]

**Pull request merged**